### PR TITLE
Liqun/bert pretrain2

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: Add bert pretrain tests

**Motivation and Context**
to cover model pretrain cases

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+from torch.utils.data import DataLoader, RandomSampler, Dataset
+import torch.distributed as dist
+
+from transformers import BertForPreTraining, BertConfig, HfArgumentParser
+
+from concurrent.futures import ProcessPoolExecutor
+
+import onnxruntime as ort
+from onnxruntime.capi.ort_trainer import ORTTrainer, IODescription, ModelDescription
+from onnxruntime.capi.ort_trainer import LossScaler
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py:27]: Are we running this on the old API too or just the New?

File: orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+    if args.local_rank == -1:
+        args.local_rank = 0
+
+    print("args.local_rank: ", args.local_rank)
+    torch.cuda.set_device(args.local_rank)
+    device = torch.device("cuda", args.local_rank)
+    args.n_gpu = 1
+
+    from onnxruntime.capi._pybind_state import set_cuda_device_id
+    set_cuda_device_id(args.local_rank)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py:275]:
```suggestion
```
This is old API. Setting `ORTTrainerOptions.device.id` already calls this function in the frontend

File: orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+    import sys
+    logger.warning("sys.argv: %s", sys.argv)
+    if len(sys.argv) >= 2 and sys.argv[1].startswith('--local_rank='):
+        # torch.parallel.launch
+        local_rank = int(sys.argv[1][len('--local_rank='):])
+        world_size = int(os.environ['WORLD_SIZE'])
+        print("torch.parallel.launch, local_rank/world_size: ", local_rank, '/', world_size)
+
+        from onnxruntime.capi._pybind_state import set_cuda_device_id
+        set_cuda_device_id(local_rank)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py:510]:
```suggestion
```

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]

**Comment by liqunfu** [comment]
I have to revert back PyTorch version. I see test failures with current PyTorch master. These failures are golden number changes. It is likely caused by data loader changes. I need to understand and fix if needed before advance PyTorch version.

**Review by thiagocrepaldi — APPROVED** [review]

**Pull request merged**